File: CodeBase/Nesi.Main/angular/src/app/models/pages/timesheet/InsertTimesheetWorkOrder.ts
import { InsertTimeSheetBase } from './InsertTimesheetBase';
import { TimesheetHourTypeRecord } from './timesheetHourType';

export interface InsertTimeSheetWorkOrder extends InsertTimeSheetBase {
  selectedCustomerId: number;
  selectedCustomerName: string;
  selectedWorkOrderId: number;
  selectedWorkOrderName: string;
  selectedTsLiteType: number;

  selectedJobType: number;
  allow_jobtype_selection: boolean;
  entry_type: string;
  timesheetHourTypeList: Array<TimesheetHourTypeRecord>;
  scope_id:number;
  mileage_value:number;
  mileage_unit:string;
  is_prevailing_wage: boolean;
}

File: CodeBase/Nesi.Main/NESI.DTO/ViewModels/Page/TimeSheet/InsertTimeSheetWorkOrder.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace NESI.DTO.ViewModels.Page.TimeSheet
{
	public class InsertTimeSheetWorkOrder : InsertTimeSheetBase
	{
		[Required]
		public int SelectedCustomerId { get; set; }
		[Required]
		public string SelectedCustomerName { get; set; }
		[Required]
		public int SelectedWorkOrderId { get; set; }
		[Required]
		public string SelectedWorkOrderName { get; set; }
		public int SelectedTsLiteType { get; set; }
        public bool is_prevailing_wage { get; set; }

        public override int CustId => SelectedCustomerId;
		public override string CustName => SelectedCustomerName;
	}
}